### PR TITLE
[SPEC] v1::Split + onnx_importer adaptation

### PR DESCRIPTION
**Pull request by tomdol**

### CODE REVIEW

File: src/ngraph/op/fused/split.cpp [hunk 1]
```diff
@@ -140,5 +140,96 @@ shared_ptr<Node> op::Split::copy_with_new_args(const NodeVector& new_args) const
...
+    case element::Type_t::i32: axis_value = axis_input->get_vector<int32_t>().at(0); break;
+    case element::Type_t::i64: axis_value = axis_input->get_vector<int64_t>().at(0); break;
+    default: break;
+    }
+#if defined(__clang__)
+#pragma clang diagnostic pop
+#endif
+
+    return axis_value;
+}
```
Comment (tsocha) [inline src/ngraph/op/fused/split.cpp]:
```suggestion
}

```

File: src/ngraph/opsets/opset0_tbl.hpp [hunk 2]
```diff
@@ -193,7 +193,6 @@ NGRAPH_OP(Softmax, ngraph::op)
 NGRAPH_OP(SoftmaxCrossEntropy, ngraph::op)
 NGRAPH_OP(SoftmaxCrossEntropyBackprop, ngraph::op)
 NGRAPH_OP(SpaceToDepth, ngraph::op)
-NGRAPH_OP(Split, ngraph::op)
```
Comment (tsocha) [inline src/ngraph/opsets/opset0_tbl.hpp]: Why `Split` has been removed from `opset0`?

Comment (tomdol) [reply src/ngraph/opsets/opset0_tbl.hpp]: That's my mistake, thanks, I'll revert it.

File: src/ngraph/pass/opset1_upgrade.cpp [hunk 3]
```diff
@@ -547,6 +547,35 @@ namespace
...
+        const auto& splits_vec = node->get_splits();
+        const auto first_elem = splits_vec.front();
+
+        const bool split_evenly =
+            std::all_of(splits_vec.begin(), splits_vec.end(), [first_elem](const size_t split) {
+                return split == first_elem;
+            });
+
+        std::shared_ptr<Node> replacement_node;
+        if (split_evenly)
```
Comment (postrational) [inline src/ngraph/pass/opset1_upgrade.cpp]: This looks like it maybe should be the other way around (`if`/`else`).

File: src/ngraph/pass/opset1_upgrade.cpp [hunk 4]
```diff
@@ -547,6 +547,35 @@ namespace
...
 
+    bool op_cast(shared_ptr<op::Split> node)
+    {
+        const auto& splits_vec = node->get_splits();
+        const auto first_elem = splits_vec.front();
+
+        const bool split_evenly =
+            std::all_of(splits_vec.begin(), splits_vec.end(), [first_elem](const size_t split) {
+                return split == first_elem;
+            });
```
[the remark this answers is not in the document]
Comment (tomdol) [reply src/ngraph/pass/opset1_upgrade.cpp]: That flag was created for internal purposes only (just for the shape inference method).

File: src/ngraph/op/fused/split.cpp [hunk 5]
```diff
@@ -56,7 +56,7 @@ op::Split::Split(const Output<Node>& data, int axis, const std::vector<size_t>&
 }
 
 // TODO REMOVE THIS CONSTRUCTOR. INTRODUCED TO PROVIDE CI COMPATIBILITY
-op::Split::Split(const Output<Node>& data, int axis, size_t num_split)
+op::v0::Split::Split(const Output<Node>& data, int axis, size_t num_split)
```
[the remark this answers is not in the document]
Comment (tomdol) [reply src/ngraph/op/fused/split.cpp]: I've removed this one, removing the other is a bit tricky but I'll give it a shot

File: src/ngraph/pass/opset1_upgrade.cpp (hunk 3 above)
Comment (tomdol) [reply src/ngraph/pass/opset1_upgrade.cpp]: done

**Pull request merged**